scripts/boxgame.js

```diff
@@ -0,0 +1,119 @@
+class BoxGameComponent {
+    box = {};
+    score = document.createElement("div");
+    squares = [];
+    container = document.createElement('div');
+    gameAreaEl = document.createElement("div");
+    gamebox = {
+        x: Math.floor(800 / 100),
+        y: Math.floor(200 / 100)
+    };
+    player = {
+        speed: 100,
+        square: 1,
+        score: 0
+    };
+
+    init = (button) => {
+        this.gameAreaEl.innerHTML = '';
+        this.container.innerHTML = '';
+        this.score.innerHTML = '';
+        document.body.appendChild(this.container);
+        this.createGameAreaDom();
+        this.addListeners();
+        this.build();
+        button.style.display = 'none';
```

Comment:
you have passed in a button element that is external to the class simply to change its display property.
is that the responsibility of this class? Maybe you want to reuse this game in a webpage where the button doesnt disappear when you click on it?
also it is best practise to assign a css class or id like '.clickedState' rather than apply the actual style in code